libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -82,11 +81,8 @@ template withEOFExceptions(body: untyped): untyped =
 
 proc cleanupTimer(s: LPChannel) {.async.} =
   ## cleanup timers
-  ##
-  if not(isNil(s.timerFut)) and
-    not(s.timerFut.finished):
-    s.timerFut.cancel()
-    await s.timerTaskFut
+  if not s.timerTaskFut.finished:
```

Comment:
can we add a test for this, since it appears to be non trivial?

Reply:
This is protection for double close which happens pretty often inside of `nim-libp2p`. `cleanupTimer()` procedure gets called from 3 different places.

For example `timeoutMonitor()` calls `reset()` which calls `cleanupTimer()`. Also somebody else could call `reset` or `close` procedures which will call `cleanupTimer()`. So this check for `timerTaskFut.finished()` only preserve you from seeing double completion Future[T] defect.

Reply:
ok, we can merge this and add the test later, this seems to be pretty straight forward after all.